tests/git.zunit

```diff
@@ -803,13 +864,58 @@
         assert ${actual2[1]} same_as 'containing'
 
         actual3=(${(0)lines[3]})
-        assert ${#actual3} equals 2
+        assert ${#actual3} equals 3
         assert ${actual3[1]} same_as 'newlines'
         assert ${actual3[2]} same_as 'file1'
+        assert ${actual3[3]} same_as 'newfile'
     }
 
     prefix=
-    _fzf_complete_git 'git reset -- '
+    _fzf_complete_git 'git reset another-branch -- '
+}
+
+@test 'Testing completion: git reset --soft -- **' {
+    _fzf_complete() {
+        fail '_fzf_complete should not be invoked'
+    }
+
+    prefix=
+    _fzf_complete_git 'git reset --soft -- '
+
+    pass
```

Comment:
We can assert that `_fzf_complete_git` has exited with 0 like the other cases in this git.zunit although it is automatically tested.
```suggestion
    assert $? equals 0
```